crates/python/src/api.rs

```diff
@@ -221,3 +234,16 @@ pub fn get_quilc_version(py: Python<'_>) -> PyResult<&PyAny> {
         Ok(version)
     })
 }
+
+#[pyfunction]
+pub fn list_quantum_processors(py: Python<'_>) -> PyResult<&PyAny> {
+    pyo3_asyncio::tokio::future_into_py(py, async move {
+        let client = Qcs::load()
+            .await
+            .map_err(|e| InvalidConfigError::new_err(e.to_string()))?;
+        let names = list_quantum_processor_names(&client, None)
+            .await
+            .map_err(|e| QcsListQuantumProcessorsError::new_err(e.to_string()))?;
```

Comment:
This method is provided by `rigetti-pyo3` and handles all of the wrapper shenanigans. Basically the same effect, but a little shorter.

```suggestion
            .map_err(ListQuantumProcessorsError::to_py_err)?;
```